src/itest/kotlin/nl/lifely/zac/itest/SignaleringenRestServiceTest.kt

```diff
@@ -0,0 +1,278 @@
+import org.json.JSONObject
+import org.mockserver.model.HttpStatusCode
+import java.time.LocalDateTime
+import java.time.ZoneId
+import java.time.ZonedDateTime
+import java.time.format.DateTimeFormatter
+import kotlin.time.Duration.Companion.milliseconds
+import kotlin.time.Duration.Companion.seconds
+
+/**
+ * This test assumes previous tests completed successfully.
+ */
+@Order(TEST_SPEC_ORDER_AFTER_TASK_COMPLETED)
+class SignaleringenRestServiceTest : BehaviorSpec() {
+    private val logger = KotlinLogging.logger {}
+    private val itestHttpClient = ItestHttpClient()
+
+    init {
+        Given("A test user") {
+            When("a dashboard notification is turned on") {
+                val notificationBodies = arrayOf(
+                    """{"dashboard":true,"mail":false,"subjecttype":"ZAAK","type":"ZAAK_DOCUMENT_TOEGEVOEGD"}""",
+                    """{"dashboard":true,"mail":false,"subjecttype":"ZAAK","type":"ZAAK_OP_NAAM"}""",
+                    """{"dashboard":true,"mail":false,"subjecttype":"ZAAK","type":"ZAAK_VERLOPEND"}""",
+                    """{"dashboard":true,"mail":false,"subjecttype":"TAAK","type":"TAAK_OP_NAAM"}"""
+                )
+                notificationBodies.forEach {
+                    val response = itestHttpClient.performPutRequest(
+                        url = "$ZAC_API_URI/signaleringen/instellingen",
+                        headers = Headers.headersOf(
+                            "Content-Type",
+                            "application/json"
+                        ),
+                        requestBodyAsString = it,
+                        addAuthorizationHeader = true
+                    )
+
+                    Then("the responses should be 'ok'") {
+                        response.code shouldBe HttpStatusCode.OK_200.code()
+                    }
+                }
+            }
+        }
+
+        Given("A created zaak") {
+            When("it is assigned to a user") {
+                val response = itestHttpClient.performPatchRequest(
+                    url = "$ZAC_API_URI/zaken/toekennen",
+                    headers = Headers.headersOf(
+                        "Content-Type",
+                        "application/json"
+                    ),
+                    requestBodyAsString = """{
+                        "zaakUUID":"$zaak1UUID",
+                        "behandelaarGebruikersnaam":"$TEST_USER_1_ID",
+                        "groepId":"$TEST_GROUP_A_ID",
+                        "reden":null
+                    }
+                    """.trimIndent()
+                )
+
+                Then("the responses should be 'ok'") {
+                    response.code shouldBe HttpStatusCode.OK_200.code()
+                }
+            }
+        }
+
+        Given("A zaak with informatie objecten") {
+            val zaakPath = "zaken/api/v1/zaken/$zaak1UUID"
+
+            val zaakInformatieObjectenResponse = itestHttpClient.performGetRequest(
```

Comment:
Why do you need to call the OpenZaak API itself in our itests? I mean: our itests should mimic the browser in that they only call the ZAC API right? Unless there is an explicit need to call other components directly of course. What is this reason in this case? Can't you get the required data from the ZAC API itself? I guess not or you would not have gone through this trouble.. ;-)

Reply:
Our APIs replace/convert information object UUID and URL. When creating a signal there are resource checks and access to OpenZaak APIs.

So I needed the original URL UUID of the inf.object and role and these were nowhere to be found in the responses we generate :(

Reply:
Ah yes, I see. Tricky indeed. 

ps: If you have the UUID you can construct the OpenZaak URL yourself if you know the syntax of those URLs. Not sure if that is feasible here?

Reply:
I know the URL, but I need a valid UUID. That's why I reach to OpenZaak to get these.